src/Dictionaries/HashedArrayDictionary.cpp

```diff
@@ -406,14 +406,19 @@ void HashedArrayDictionary<dictionary_key_type>::createAttributes()
 template <DictionaryKeyType dictionary_key_type>
 void HashedArrayDictionary<dictionary_key_type>::updateData()
 {
-    if (!update_field_loaded_block || update_field_loaded_block->rows() == 0)
+    if (!update_field_loaded_block)
     {
         QueryPipeline pipeline(source_ptr->loadUpdatedAll());
-
         PullingPipelineExecutor executor(pipeline);
         Block block;
+
         while (executor.pull(block))
         {
+            if (!block.rows())
+                continue;
```

Comment:
It seems some of the blocks returned by `executor.pull()` were empty and as a result `update_field_loaded_block = std::make_shared<DB::Block>(block.cloneEmpty());` below could clone an empty block and after that at any next iteration it could try to add more rows to that empty block (see the call `saved_column->insertRangeFrom(...)` below), which was wrong.

Reply:
Any data obtained from `source_ptr` is added to `update_field_loaded_block`. In order to do that they must have the same structure (number of columns and data types). If data obtained from `source_ptr` has no columns and we create `update_field_loaded_block` from that data then `update_field_loaded_block` will also have no columns, making impossible to add following blocks with columns to that `update_field_loaded_block` without columns.